amazon/ion/writer_text.py

```diff
@@ -173,15 +173,33 @@ def _bytes_datetime(dt):
         return tz_string + _bytes_utc_offset(dt)
 
     fractional_precision = getattr(original_dt, TIMESTAMP_FRACTION_PRECISION_FIELD, MICROSECOND_PRECISION)
-    if fractional_precision:
+    fractional_seconds = getattr(original_dt, TIMESTAMP_FRACTIONAL_SECONDS_FIELD, None)
+    if fractional_precision is not None and fractional_seconds is None:
         fractional = dt.strftime('%f')
         assert len(fractional) == MICROSECOND_PRECISION
+
         if fractional[fractional_precision:] != ('0' * (MICROSECOND_PRECISION - fractional_precision)):
             raise ValueError('Found timestamp fractional with more than the specified %d digits of precision.'
                              % (fractional_precision,))
         fractional = fractional[:fractional_precision]
         tz_string += '.' + fractional
-
+    else:
```

Comment:
Should this be `elif fractional_seconds is not None:`?

Reply:
Done.